src/documents/forms/EditDocumentForm.js

```diff
@@ -49,10 +50,29 @@ const EditDocumentForm = React.forwardRef(
             file_desc: fileDescription,
             file_status: versionStatus,
           }}
-          validate={vals => validate(vals, fileNode, studyFiles)}
+          validate={vals => {
+            let errors = {};
+            setFileNameError(validate(vals, fileNode, studyFiles));
+            let checkedFields = ['file_name', 'file_type', 'file_desc'];
+            checkedFields.forEach(function(field) {
+              if (!vals[field]) {
+                errors[field] = true;
+              }
+            });
+            return errors;
+          }}
           onSubmit={values => {
-            setUploading(true);
-            handleSubmit(...Object.values(values));
+            if (
+              !(
+                !isAdmin &&
```

Comment:
doesn't this block non-admin users from updating file names?

Reply:
in this case, for the regular user, file name has to be valid before saving, for admin user, they can save the file name even when it's not valid